api/groups.go

```diff
@@ -11,6 +11,44 @@ import (
 	"github.com/gorilla/mux"
 )
 
+//CreateGroupHandler creates a new group
+func (api *API) CreateGroupHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, handleBodyReadError(ctx, err)
+	}
+
+	createGroup := models.CreateGroup{}
+	err = json.Unmarshal(body, &createGroup)
+	if err != nil {
+		return nil, handleBodyUnmarshalError(ctx, err)
+	}
+
+	validationErrs := createGroup.ValidateCreateGroupRequest(ctx)
+	if len(validationErrs) != 0 {
+		return nil, models.NewErrorResponse(http.StatusBadRequest, nil, validationErrs...)
+	}
+
+	// build create group input
+	createGroup.GenerateGroupName()
+	input := createGroup.BuildCreateGroupInput(&api.UserPoolId)
+	_, err = api.CognitoClient.CreateGroup(input)
+	if err != nil {
+		cognitoErr := models.NewCognitoError(ctx, err, "Cognito CreateGroup request from create a new group endpoint")
+		if cognitoErr.Code == models.GroupExistsError {
```

Comment:
Are there any other scenarios that should return BadRequest? (e.g. invalid precedence value)
Or does the internal validation cover all of these?

Reply:
Yep, internal validation covers all these scenarios.

Reply:
The scenario I thought wasn't covered was if the precedence was over the max, but just reread the docs and there is no max value.
Not sure where I got the max of 100 from.